src/databricks/labs/ucx/source_code/files.py

```diff
@@ -150,3 +150,29 @@ def resolve_import(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
 
     def __repr__(self):
         return "LocalFileResolver()"
+
+
+class SitePackageResolver(BaseDependencyResolver):
+
+    def __init__(
+        self,
+        file_loader: FileLoader,
+        site_packages_path: Path,
+        next_resolver: BaseDependencyResolver | None = None,
+    ):
+        super().__init__(next_resolver)
+        self._file_loader = file_loader
+        self._site_packages_path = site_packages_path
+
+    def with_next_resolver(self, resolver: BaseDependencyResolver) -> BaseDependencyResolver:
+        return SitePackageResolver(self._file_loader, self._site_packages_path, resolver)
+
+    def resolve_import(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
```

Comment:
Should this work for submodules as well? `databricks.labs.ucx`

Reply:
Should `SitePackage` not resolve a library instead? Why an import? When do we use which?

Reply:
Decision on naming needs to be made for import vs library

Reply:
Fixed submodule import